src/components/timers/Timer.tsx

```diff
@@ -0,0 +1,169 @@
+    );
+    const [timeDisplay, setTimeDisplay] = useState(
+        type === 'countdown' ? duration : 0
+    );
+
+    useEffect(() => {
+        if (state === 'completed') {
+            setCurrentRound(rounds); // Set currentRound to the maximum when completed
+        }
+    }, [state, rounds]);
+
+    // Reset timer state on READY
+    useEffect(() => {
+        if (timerStatus === TimerStatus.READY) {
+            setCurrentRound(1);
+            setIsWorkInterval(true);
+            setTimeDisplay(type === 'countdown' ? duration : 0);
+            setTotalTime(
+                type === 'stopwatch' ? duration :
+                    type === 'xy' ? roundTime :
+                        type === 'tabata' ? workTime : 0
+            );
+        }
+    }, [timerStatus, duration, type, roundTime, workTime]);
+
+    useEffect(() => {
+        if (isActive && timerStatus === TimerStatus.RUNNING) {
+            setTimeDisplay(
+                type === 'countdown'
+                    ? Math.max(duration - globalTimer, 0) // Ensure countdown doesn't go negative
+                    : globalTimer // Stopwatch displays the elapsed time
+            );
+
+            // Countdown logic
+            if (type === 'countdown' && globalTimer >= duration) {
+                timerComplete();
+            }
+
+            // Stopwatch logic
+            if (type === 'stopwatch' && globalTimer >= duration) {
+                timerComplete();
+            }
+
+            // XY Timer logic
+            if (type === 'xy') {
+                if (roundTime - globalTimer <= 0 && currentRound <= rounds) {
+                    setCurrentRound((prev) => prev + 1);
+                    resetTimer();
+                }
+                if (currentRound > rounds) {
+                    timerComplete();
+                }
+            }
+
+            // Tabata Timer logic
+            if (type === 'tabata') {
+                if (isWorkInterval && globalTimer >= workTime) {
+                    setIsWorkInterval(false);
+                    resetTimer();
+                } else if (!isWorkInterval && globalTimer >= restTime) {
+                    setIsWorkInterval(true);
+                    setCurrentRound((prev) => prev + 1);
+                    resetTimer();
+                }
+                if (currentRound > rounds) {
+                    timerComplete();
+                }
+            }
+        }
+    }, [
+        isActive,
+        timerStatus,
+        globalTimer,
+        duration,
+        type,
+        currentRound,
+        rounds,
+        roundTime,
+        workTime,
+        restTime,
+        isWorkInterval,
+        timerComplete,
+        resetTimer,
+    ]);
+
+
+
+    const title = {
+        stopwatch: 'Stopwatch',
+        countdown: 'Countdown Timer',
+        xy: 'XY Timer',
+        tabata: 'Tabata Timer',
+    }[type];
+
+    if (!title) {
+        return <div>Invalid timer type.</div>;
+    }
+
+    return (
```

Comment:
I'd recommend visually highlighting the timer in some way when it is active (ex. big red border). If someone is working out, it will help draw their eye to which timer is currently running.